linetools/lists/linelist.py

```diff
@@ -493,16 +526,23 @@ def subset_lines(self, subset, reset_data=False, verbose=False, sort_by=['wrest'
         """
 
         # Check the right format
-        if not isinstance(subset, (list, np.ndarray)):
-            raise ValueError('subset_lines: the input subset must be a list!')
+        if not isinstance(subset, (list, Quantity)):
+            raise ValueError('subset_lines: the input subset must be a list or Quantity array!')
 
         # Reset _data (useful for changing subsets)
         if reset_data:
             self.set_lines(verbose=False)
 
         indices = []
-        if isinstance(subset[0], (float, Quantity)):  # wrest
-            wrest = self._data['wrest'] # Assumes Angstroms
+        if isinstance(subset, Quantity):  # wrest
+            wrest = self._data['wrest'].to('AA').value # Assumes Angstroms
```

Comment:
remove inline comment. this is not longer an assuption because we convert to `AA` (otherwise it will raise an error).

Reply:
removed